src/Bookings.js

```diff
@@ -1,18 +1,77 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import CustomerProfile from "./CustomerProfile.js";
 import Search from "./Search.js";
-// import SearchResults from "./SearchResults.js";
-// import FakeBookings from "./data/fakeBookings.json";
+import SearchResults from "./SearchResults.js";
 
-const Bookings = () => {
+const Bookings = props => {
+  const [bookings, setBookings] = useState([]);
+  const [profileId, setProfileId] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+
+  const onShowProfile = id => {
+    setProfileId(id);
+  };
+
+  const resetCustomerProfileId = () => {
+    setProfileId(null);
+  };
+
+  useEffect(() => {
+    const fetchBookings = async () => {
+      const res = await fetch("https://cyf-react.glitch.me/delayed");
+      /* fetch(`https://cyf-react.glitch.me/error`) */
+
+      if (res.ok) {
+        const data = await res.json();
+        setBookings(data);
+        setLoading(false);
+      } else {
+        setLoading(false);
+        setError(true);
+      }
+    };
+
+    fetchBookings();
+  }, [props.bookings]);
+  if (loading) {
+    return (
+      <div className="text-center">
+        <div className="spinner-border" role="status">
+          <span className="sr-only">Loading...</span>
+        </div>
+        <p>Loading...</p>
+      </div>
+    );
+  }
+  if (error) {
+    return (
+      <div className="centered">
+        <h1>500 Server Error</h1>
+        <p>It's broken, but it's not your fault!</p>
+      </div>
+    );
+  }
   const search = searchVal => {
-    console.info("TO DO!", searchVal);
+    const filterSearch = bookings.filter(
+      nameInput =>
+        nameInput.firstName.toLowerCase().includes(searchVal) ||
+        nameInput.surname.toLowerCase().includes(searchVal)
+    );
+    setBookings(filterSearch);
   };
```

Comment:
really nice that you've thought about catching lowercase searches! At the moment, if I type in 'John', it doesn't come up with anything - could add an extra step in here to compare lowercase with lowercase